nohup: ignoring input
/home/dgxuser16/.local/lib/python3.8/site-packages/opacus/privacy_engine.py:95: UserWarning: Secure RNG turned off. This is perfectly fine for experimentation as it allows for much faster training performance, but remember to turn it on and retrain one last time before production with ``secure_mode`` turned on.
  warnings.warn(
01/25/2025 03:53:14:WARNING:Ignoring drop_last as it is not compatible with DPDataLoader.
01/25/2025 03:53:14:DEBUG:Opened insecure gRPC connection (no certificates were passed)
01/25/2025 03:53:14:DEBUG:ChannelConnectivity.IDLE
01/25/2025 03:53:14:DEBUG:ChannelConnectivity.CONNECTING
01/25/2025 03:53:14:DEBUG:ChannelConnectivity.READY
[92mINFO [0m:      
01/25/2025 03:53:54:INFO:
[92mINFO [0m:      [RUN 0, ROUND ]
01/25/2025 03:53:54:INFO:[RUN 0, ROUND ]
[92mINFO [0m:      Received: train message 7f184ce7-4a98-4401-9a09-76b76c7bd4b0
01/25/2025 03:53:54:INFO:Received: train message 7f184ce7-4a98-4401-9a09-76b76c7bd4b0
/home/dgxuser16/.local/lib/python3.8/site-packages/torch/nn/modules/module.py:1359: UserWarning: Using a non-full backward hook when the forward contains multiple autograd Nodes is deprecated and will be removed in future versions. This hook will be missing some grad_input. Please use register_full_backward_hook to get the documented behavior.
  warnings.warn("Using a non-full backward hook when the forward contains multiple autograd Nodes "
[92mINFO [0m:      Sent reply
01/25/2025 03:58:36:INFO:Sent reply
[92mINFO [0m:      
01/25/2025 03:59:21:INFO:
[92mINFO [0m:      [RUN 0, ROUND ]
01/25/2025 03:59:21:INFO:[RUN 0, ROUND ]
[92mINFO [0m:      Received: evaluate message adc6ab74-86da-4d08-bc12-b98b3bcd8928
01/25/2025 03:59:21:INFO:Received: evaluate message adc6ab74-86da-4d08-bc12-b98b3bcd8928
[92mINFO [0m:      Sent reply
01/25/2025 03:59:53:INFO:Sent reply
[92mINFO [0m:      
01/25/2025 04:00:36:INFO:
[92mINFO [0m:      [RUN 0, ROUND ]
01/25/2025 04:00:36:INFO:[RUN 0, ROUND ]
[92mINFO [0m:      Received: train message 43f4dd6c-b095-45b8-adf3-0e2deec271c5
01/25/2025 04:00:36:INFO:Received: train message 43f4dd6c-b095-45b8-adf3-0e2deec271c5
/home/dgxuser16/.local/lib/python3.8/site-packages/torch/nn/modules/module.py:1359: UserWarning: Using a non-full backward hook when the forward contains multiple autograd Nodes is deprecated and will be removed in future versions. This hook will be missing some grad_input. Please use register_full_backward_hook to get the documented behavior.
  warnings.warn("Using a non-full backward hook when the forward contains multiple autograd Nodes "
[92mINFO [0m:      Sent reply
01/25/2025 04:04:38:INFO:Sent reply
[92mINFO [0m:      
01/25/2025 04:05:25:INFO:
[92mINFO [0m:      [RUN 0, ROUND ]
01/25/2025 04:05:25:INFO:[RUN 0, ROUND ]
[92mINFO [0m:      Received: evaluate message fa910817-8595-4d38-8e0d-74d767295075
01/25/2025 04:05:25:INFO:Received: evaluate message fa910817-8595-4d38-8e0d-74d767295075
[92mINFO [0m:      Sent reply
01/25/2025 04:06:03:INFO:Sent reply
[92mINFO [0m:      
01/25/2025 04:06:37:INFO:
[92mINFO [0m:      [RUN 0, ROUND ]
01/25/2025 04:06:37:INFO:[RUN 0, ROUND ]
[92mINFO [0m:      Received: train message 50e77d15-93ee-4684-8e19-f8efbc82612d
01/25/2025 04:06:37:INFO:Received: train message 50e77d15-93ee-4684-8e19-f8efbc82612d
/home/dgxuser16/.local/lib/python3.8/site-packages/torch/nn/modules/module.py:1359: UserWarning: Using a non-full backward hook when the forward contains multiple autograd Nodes is deprecated and will be removed in future versions. This hook will be missing some grad_input. Please use register_full_backward_hook to get the documented behavior.
  warnings.warn("Using a non-full backward hook when the forward contains multiple autograd Nodes "
[91mERROR [0m:     Client raised an exception.
Traceback (most recent call last):
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/flwr/client/app.py", line 496, in _start_client_internal
    reply_message = client_app(message=message, context=context)
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/flwr/client/client_app.py", line 98, in __call__
    return self._call(message, context)
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/flwr/client/client_app.py", line 81, in ffn
    out_message = handle_legacy_message_from_msgtype(
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/flwr/client/message_handler/message_handler.py", line 130, in handle_legacy_message_from_msgtype
    fit_res = maybe_call_fit(
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/flwr/client/client.py", line 234, in maybe_call_fit
    return client.fit(fit_ins)
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/flwr/client/numpy_client.py", line 238, in _fit
    results = self.numpy_client.fit(parameters, ins.config)  # type: ignore
  File "client_4.py", line 147, in fit
    epsilon = train(
  File "client_4.py", line 85, in train
    optimizer.step()
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/opacus/optimizers/optimizer.py", line 518, in step
    if self.pre_step():
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/opacus/optimizers/optimizer.py", line 499, in pre_step
    self.clip_and_accumulate()
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/opacus/optimizers/optimizer.py", line 412, in clip_and_accumulate
    grad = contract("i,i...", per_sample_clip_factor, grad_sample)
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/opt_einsum/contract.py", line 507, in contract
    return _core_contract(operands, contraction_list, backend=backend, **einsum_kwargs)
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/opt_einsum/contract.py", line 573, in _core_contract
    new_view = _tensordot(*tmp_operands, axes=(tuple(left_pos), tuple(right_pos)), backend=backend)
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/opt_einsum/sharing.py", line 131, in cached_tensordot
    return tensordot(x, y, axes, backend=backend)
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/opt_einsum/contract.py", line 374, in _tensordot
    return fn(x, y, axes=axes)
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/opt_einsum/backends/torch.py", line 54, in tensordot
    return torch.tensordot(x, y, dims=axes)
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/torch/functional.py", line 1193, in tensordot
    return _VF.tensordot(a, b, dims_a, dims_b)  # type: ignore[attr-defined]
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu! (when checking argument for argument mat2 in method wrapper_CUDA_mm)
01/25/2025 04:09:18:ERROR:Client raised an exception.
Traceback (most recent call last):
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/flwr/client/app.py", line 496, in _start_client_internal
    reply_message = client_app(message=message, context=context)
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/flwr/client/client_app.py", line 98, in __call__
    return self._call(message, context)
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/flwr/client/client_app.py", line 81, in ffn
    out_message = handle_legacy_message_from_msgtype(
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/flwr/client/message_handler/message_handler.py", line 130, in handle_legacy_message_from_msgtype
    fit_res = maybe_call_fit(
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/flwr/client/client.py", line 234, in maybe_call_fit
    return client.fit(fit_ins)
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/flwr/client/numpy_client.py", line 238, in _fit
    results = self.numpy_client.fit(parameters, ins.config)  # type: ignore
  File "client_4.py", line 147, in fit
    epsilon = train(
  File "client_4.py", line 85, in train
    optimizer.step()
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/opacus/optimizers/optimizer.py", line 518, in step
    if self.pre_step():
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/opacus/optimizers/optimizer.py", line 499, in pre_step
    self.clip_and_accumulate()
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/opacus/optimizers/optimizer.py", line 412, in clip_and_accumulate
    grad = contract("i,i...", per_sample_clip_factor, grad_sample)
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/opt_einsum/contract.py", line 507, in contract
    return _core_contract(operands, contraction_list, backend=backend, **einsum_kwargs)
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/opt_einsum/contract.py", line 573, in _core_contract
    new_view = _tensordot(*tmp_operands, axes=(tuple(left_pos), tuple(right_pos)), backend=backend)
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/opt_einsum/sharing.py", line 131, in cached_tensordot
    return tensordot(x, y, axes, backend=backend)
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/opt_einsum/contract.py", line 374, in _tensordot
    return fn(x, y, axes=axes)
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/opt_einsum/backends/torch.py", line 54, in tensordot
    return torch.tensordot(x, y, dims=axes)
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/torch/functional.py", line 1193, in tensordot
    return _VF.tensordot(a, b, dims_a, dims_b)  # type: ignore[attr-defined]
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu! (when checking argument for argument mat2 in method wrapper_CUDA_mm)
01/25/2025 04:09:19:DEBUG:gRPC channel closed
Epsilon = 0.53

{'loss': [175.82146698236465], 'accuracy': [0.49960906958561374], 'auc': [0.38544397756630056]}

Epsilon = 0.78

{'loss': [175.82146698236465, 175.51212838292122], 'accuracy': [0.49960906958561374, 0.4988272087568413], 'auc': [0.38544397756630056, 0.4424269186429919]}

Traceback (most recent call last):
  File "client_4.py", line 183, in <module>
    fl.client.start_client(
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/flwr/client/app.py", line 291, in start_client
    _start_client_internal(
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/flwr/client/app.py", line 503, in _start_client_internal
    raise ex
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/flwr/client/app.py", line 496, in _start_client_internal
    reply_message = client_app(message=message, context=context)
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/flwr/client/client_app.py", line 98, in __call__
    return self._call(message, context)
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/flwr/client/client_app.py", line 81, in ffn
    out_message = handle_legacy_message_from_msgtype(
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/flwr/client/message_handler/message_handler.py", line 130, in handle_legacy_message_from_msgtype
    fit_res = maybe_call_fit(
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/flwr/client/client.py", line 234, in maybe_call_fit
    return client.fit(fit_ins)
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/flwr/client/numpy_client.py", line 238, in _fit
    results = self.numpy_client.fit(parameters, ins.config)  # type: ignore
  File "client_4.py", line 147, in fit
    epsilon = train(
  File "client_4.py", line 85, in train
    optimizer.step()
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/opacus/optimizers/optimizer.py", line 518, in step
    if self.pre_step():
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/opacus/optimizers/optimizer.py", line 499, in pre_step
    self.clip_and_accumulate()
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/opacus/optimizers/optimizer.py", line 412, in clip_and_accumulate
    grad = contract("i,i...", per_sample_clip_factor, grad_sample)
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/opt_einsum/contract.py", line 507, in contract
    return _core_contract(operands, contraction_list, backend=backend, **einsum_kwargs)
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/opt_einsum/contract.py", line 573, in _core_contract
    new_view = _tensordot(*tmp_operands, axes=(tuple(left_pos), tuple(right_pos)), backend=backend)
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/opt_einsum/sharing.py", line 131, in cached_tensordot
    return tensordot(x, y, axes, backend=backend)
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/opt_einsum/contract.py", line 374, in _tensordot
    return fn(x, y, axes=axes)
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/opt_einsum/backends/torch.py", line 54, in tensordot
    return torch.tensordot(x, y, dims=axes)
  File "/home/dgxuser16/.local/lib/python3.8/site-packages/torch/functional.py", line 1193, in tensordot
    return _VF.tensordot(a, b, dims_a, dims_b)  # type: ignore[attr-defined]
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu! (when checking argument for argument mat2 in method wrapper_CUDA_mm)
